api/turing/models/traffic_rules.go

```diff
@@ -10,7 +10,7 @@ import (
 
 type TrafficRule struct {
 	Name       string                         `json:"name" validate:"required,notBlank"`
-	Conditions []*router.TrafficRuleCondition `json:"conditions" validate:"required,notBlank,dive"`
+	Conditions []*router.TrafficRuleCondition `json:"conditions" validate:"required,dive"`
```

Comment:
We can now put back the `notBlank` here, no?

Reply:
We can't because we're using this struct in `api/turing/cluster/servicebuilder/router.go` (https://github.com/caraml-dev/turing/pull/232/files#diff-b42915e302303b3cc6b675a5bbbb46862a411038f37f43d836b82037892bbf23R515). If we add `notBlank` here, we won't be able to set an empty conditions slice for Default Traffic Rule.

Reply:
Technically, we can add the empty slice in this step right? There is no validation triggered afterwards? AFAIK, the only time this struct is used in validation is during handling of the API call to create / edit router (where we expect things to be `notBlank`). Am I missing something?

Reply:
One of the benefits of splitting the default traffic rule into a new field is also easier separation of the validation. If we are unable to set `notBlank` on that struct because of the step where we merge the rules for the Fiber config, I think we can resort to defining a separate struct for that phase.

Reply:
>There is no validation triggered afterwards? AFAIK, the only time this struct is used in validation is during handling of the API call to create / edit router (where we expect things to be notBlank). 

This is correct and I got what you mean. I confused this field with another, and I've added `notBlank` tag back.